postprocessors/sandbox/widebeam_downsample2normalscan.py

```diff
@@ -0,0 +1,217 @@
+                raise conversion_exceptions.ConversionUpstreamError(
+                    f'Input file structure "{self.infile_structure}" cannot be reprocessed into any other format.'
+                )
+        check_args(self)
+        self.averaging_method = None
+        self._temp_files = []
+        self.process_file()
+
+    @staticmethod
+    def process_record(record: OrderedDict, **kwargs) -> OrderedDict:
+        """
+        Takes a record from a rawacf file process into a rawacf record.
+        This method also keeps a single designated beam,
+
+        Parameters
+        ----------
+        record: OrderedDict
+            hdf5 record containing rawacf data and metadata
+        beam_index: Dict
+            A dictionary mapping timestamp to beam index
+
+        Returns
+        -------
+        record: OrderedDict
+            hdf5 record, downsampled to one beam
+        """
+        beam2keep = 0
+        first_min = dt.datetime.fromtimestamp(record['sqn_timestamps'][0]).replace(second =0, microsecond = 0)
+        timestamp = dt.datetime.fromtimestamp(record['sqn_timestamps'][0])
+        diff = abs(first_min - timestamp).total_seconds()/record['int_time']
+        beam2keep = int(round(diff))
+        if beam2keep >= 16:
+            beam2keep = 0
+
+        #Seperate the beam to keep
+        record['beam_nums'] = np.array([np.uint32(beam2keep)])
+        record['beam_azms'] = np.array([record['beam_azms'][beam2keep]])
+        try:
+            record['data_dimensions'][0] = 1
+        except:
+            record['correlation_dimensions'][0] = 1
+        record['main_acfs'] = record['main_acfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        record['intf_acfs'] = record['intf_acfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        record['xcfs'] = record['xcfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        if beam2keep == 0:
+            record['scan_start_marker'] = True
+        else:
+            record['scan_start_marker'] = False
+        return record
+
+    @staticmethod
+    def dmap_to_dmap(file_to_process: str, processed_file: str, **kwargs) -> dict:
+        """
+        Checks what beam index is associated to timestamp and keeps only that beam from a set of 16 records.
+
+        Parameters
+        ----------
+        all_records: list
+            list of timestamps
+        record: dict
+            dictionary containing rawacf data, where the key is timestamp and each entry is a list of 16 beams
+            corresponding to one integration time/full FOV
+
+        Returns
+        -------
+        beamedrec: list
+            The downsampled record
+        """
+
+        sdarn_read = pydarnio.SDarnRead(file_to_process)
+        data = sdarn_read.read_rawacf()
+
+        record = dict()
+        all_records = []  # record names
```

Comment:
```suggestion
        grouped_records = []
```
With the suggestions below, `all_records` is no longer used.